src/System.Private.Uri/src/System/Uri.cs

```diff
@@ -5045,6 +5220,233 @@ private static string PathDifference(string path1, string path2, bool compareCas
+            // It does not make much sense and violates Fxcop on calling a virtual method in the ctor.
+            // Should be deprecated and removed asap.
+        }
+
+        /// <internalonly/>
+        [Obsolete("The method has been deprecated. It is not used by the system. http://go.microsoft.com/fwlink/?linkid=14202")]
+        protected virtual void Parse()
+        {
+            // this method if suppressed by the derived class
+            // would lead to an unconstructed Uri instance.
+            // It does not make any sense and violates Fxcop on calling a virtual method in the ctor.
+            // Should be deprecated and removed asap.
+        }
+
+        /// <internalonly/>
+        [Obsolete("The method has been deprecated. It is not used by the system. http://go.microsoft.com/fwlink/?linkid=14202")]
+        protected virtual void Escape()
+        {
+            // this method if suppressed by the derived class
+            // would lead to the same effect as dontEscape=true.
+            // It does not make much sense and violates Fxcop on calling a virtual method in the ctor.
+            // Should be deprecated and removed asap.
+        }
+
+        //
+        // Unescape
+        //
+        //  Convert any escape sequences in <path>. Escape sequences can be
+        //  hex encoded reserved characters (e.g. %40 == '@') or hex encoded
+        //  UTF-8 sequences (e.g. %C4%D2 == 'Latin capital Ligature Ij')
+        //
+        /// <internalonly/>
+        [Obsolete("The method has been deprecated. Please use GetComponents() or static UnescapeDataString() to unescape a Uri component or a string. http://go.microsoft.com/fwlink/?linkid=14202")]
+        protected virtual string Unescape(string path)
+        {
+            // This method is dangerous since it gives path unescaping control
+            // to the derived class without any permission demand.
+            // Should be deprecated and removed asap.
+
+            char[] dest = new char[path.Length];
+            int count = 0;
+            dest = UriHelper.UnescapeString(path, 0, path.Length, dest, ref count, c_DummyChar, c_DummyChar,
+                c_DummyChar, UnescapeMode.Unescape | UnescapeMode.UnescapeAll, null, false);
+            return new string(dest, 0, count);
+        }
+
+        [Obsolete("The method has been deprecated. Please use GetComponents() or static EscapeUriString() to escape a Uri component or a string. http://go.microsoft.com/fwlink/?linkid=14202")]
+        protected static string EscapeString(string str)
+        {
+            // This method just does not make sense as protected
+            // It should go public static asap
+
+            if ((object)str == null)
+            {
+                return string.Empty;
+            }
+
+            int destStart = 0;
+            char[] dest = UriHelper.EscapeString(str, 0, str.Length, null, ref destStart, true, '?', '#', '%');
+            if ((object)dest == null)
```

Comment:
@stephentoub My understanding is it returns null if nothing has to be escaped in str (AND passed dest was null, which it is in this case)